apps/election-manager/src/components/ExportFinalResultsModal.tsx

```diff
@@ -0,0 +1,299 @@
+        cvrFile.name,
+        new File([CastVoteRecordsString], 'cvrs')
+      )
+      await client.process()
+
+      // download the result
+      const results = await client.getOutputFile(resultsFile.name)
+      if (window.kiosk) {
+        const usbPath = await getDevicePath()
+        const pathToFile = path.join(usbPath!, `${filename}.csv`)
+        await window.kiosk!.writeFile(pathToFile, await results.text())
+      } else {
+        // For internal testing only if not in kiosk mode just download the file.
+        fileDownload(results, `${filename}.csv`, 'text/csv')
+      }
+
+      // reset server files
+      await client.reset()
+      setCurrentState(ModalState.DONE)
+    } catch (error) {
+      setErrorMessage(`Failed to save results. ${error.message}`)
+      setCurrentState(ModalState.ERROR)
+    }
+  }
+
+  if (currentState === ModalState.ERROR) {
+    return (
+      <Modal
+        isOpen={isOpen}
+        content={
+          <Prose>
+            <h1>Download Failed</h1>
+            <p>{errorMessage}</p>
+          </Prose>
+        }
+        onOverlayClick={onClose}
+        actions={<LinkButton onPress={onClose}>Close</LinkButton>}
+      />
+    )
+  }
+
+  if (currentState === ModalState.DONE) {
+    let actions = (
+      <React.Fragment>
+        <LinkButton onPress={onClose}>Close</LinkButton>
+        <USBControllerButton small={false} primary />
+      </React.Fragment>
+    )
+    if (usbDriveStatus === UsbDriveStatus.recentlyEjected) {
+      actions = <LinkButton onPress={onClose}>Close</LinkButton>
+    }
+    return (
+      <Modal
+        isOpen={isOpen}
+        content={
+          <Prose>
+            <h1>Download Complete</h1>
+            <p>
+              Results file succesfully saved as <strong>{filename}.csv</strong>{' '}
+              on the inserted USB drive. You may now eject the USB and take it
+              to the proper machine for election night reporting.
+            </p>
+          </Prose>
+        }
+        onOverlayClick={onClose}
+        actions={actions}
+      />
+    )
+  }
+
+  if (currentState === ModalState.SAVING) {
+    return (
+      <Modal isOpen={isOpen} content={<Loading />} onOverlayClick={onClose} />
+    )
+  }
+
+  if (currentState !== ModalState.INIT) {
+    throwBadStatus(currentState) // Creates a compile time check that all states are being handled.
+  }
+
+  switch (usbDriveStatus) {
+    case UsbDriveStatus.absent:
+    case UsbDriveStatus.notavailable:
+    case UsbDriveStatus.recentlyEjected:
+      // When run not through kiosk mode let the user download the file
+      // on the machine for internal debugging use
+      return (
+        <Modal
+          isOpen={isOpen}
+          content={
+            <Prose>
+              <h1>No USB Drive Detected</h1>
+              <p>
+                <USBImage src="usb-drive.svg" alt="Insert USB Image" />
+                Please insert a USB drive in order to export the scanner
+                results.
```

Comment:
```suggestion
                Please insert a USB drive where the election results will be saved.
```

Reply:
Or `Insert a USB drive to save the election results file on.` ?